sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetOptions.scala

```diff
@@ -42,8 +43,15 @@ private[parquet] class ParquetOptions(
    * Acceptable values are defined in [[shortParquetCompressionCodecNames]].
    */
   val compressionCodecClassName: String = {
```

Comment:
Can we change `compressionCodecClassName` to `compressionCodec` instead?

Reply:
@HyukjinKwon Seems you're right.
@gatorsmile  Are we mistaken, shouldn't we change ParquetOptions's  `compressionCodec ` to `compressionCodecClassName `? Because `OrcOptions` and `TextOptions` are all using `compressionCodec `.

Reply:
`compressionCodecClassName ` is a better name. We should change all the others to this.

Reply:
We could alternatively say `compressionCodecName` here. It's rather names like `UNCOMPRESSED`, `LZO`, etc in this case. For the text based sources, they are canonical class names so I am okay with `compressionCodecClassName`  but for ORC and Parquet these are not classes.

Reply:
`compressionCodecName ` is also fine to me.

Reply:
So, change all `compressionCodecClassName` and `compressionCodec` to `compressionCodecName`?  In `TextOptions` ,`JSONOptions` and `CSVOptions` too ?

Reply:
@gatorsmile @HyukjinKwon 
In `TextOptions` ,`JSONOptions` and `CSVOptions`, it's  "Option[String]", but in `OrcOptions` and `ParquetOptions`, it's a "String".
Just change `compressionCodecClassName` in `OrcOptions` and `ParquetOptions` to `compressionCodecName` is ok ?

Reply:
Let's do Parquet and ORC ones here for now if that's also fine to @gatorsmile.